fdbserver/storageserver.actor.cpp

```diff
@@ -5838,6 +5857,10 @@ void ApplyMutationsFromCursorMessageHandler::mutationRefHandler(const MutationRe
 
 	::details::updateMutationTypeCounter(pStorageServerContext, mutationRef.type);
 
+	if (mutationRef.param1 == pStorageServerContext->storageServerToTeamIDKey) {
+		pStorageServerContext->storageServerStorageTeams = StorageServerStorageTeams(mutationRef.param2);
+	}
```

Comment:
Yeah, I am aware of this, however, the main issue here is that we have two different `StorageServer` classes, while `applyPrivateData` only accepts the base class of both at this stage. So there are two possible approaches:

1. Put a `dynamic_cast` clause in the `applyPrivateData` function. This will make the already bloated `applyPrivateData` even worse.
2. Do a ptxn subclass for `StorageUpdater`. This would be much cleaner but takes time.

I think at this stage we can 1) prove this works while place a Radar ticket for the refactoring 2) do the refactoring later when ptxn works, at least cycletest works.